libjolie/src/jolie/lang/parse/OLParser.java

```diff
@@ -811,8 +812,160 @@ private void parseInterfaceOrPort()
+    {
+        //only proceed if a service declaration
+        if (!token.isKeyword("service")) return;
+        
+		//get service name
+        getToken();
+        assertToken(Scanner.TokenType.ID, "expected service name");
+        String serviceName = token.content();
+        
+        //validate token
+        getToken();
+        eat(Scanner.TokenType.LCURLY, "{ expected");
+        
+        //initialize internal interface and interface list
+        List< InterfaceDefinition > interfaceList = new ArrayList< InterfaceDefinition >();
+        
+        OLSyntaxNode internal_main = null;
+        SequenceStatement internal_init = null;
+        
+        boolean keepRunning = true;
+        while ( keepRunning ) {
+            if ( token.isKeyword( "Interfaces" ) ) {
+                getToken();
+                eat( Scanner.TokenType.COLON, "expected : after Interfaces" );
+                boolean keepRun = true;
+                while( keepRun ) {
+                    assertToken( Scanner.TokenType.ID, "expected interface name" );
+                    InterfaceDefinition i = interfaces.get( token.content() );
+                    if ( i == null ) {
+                        throwException( "Invalid interface name: " + token.content() );
+                    }
+                    interfaceList.add( i );
+                    getToken();
+
+                    if ( token.is( Scanner.TokenType.COMMA ) ) {
+                        getToken();
+                    } else {
+                        keepRun = false;
+                    }
+                }
+            }
+            else if ( token.isKeyword( "main" ) ) {
+				if ( internal_main != null ) {
+					throwException( "you must specify only one main definition" );
+				}
 
+				internal_main = parseMain();
+			} else if ( token.is( Scanner.TokenType.INIT ) ) {
+				if ( internal_init == null ) {
+					internal_init = new SequenceStatement( getContext() );
+				}
+
+				internal_init.addChild(parseInit());
+            }
+            else if (token.is(Scanner.TokenType.RCURLY)) {
+                keepRunning = false;
+            }
+            else {
+                throwException( "Unrecognized token in inline service." );
+            }
+        }
+        
+        //validate ending
+        eat(Scanner.TokenType.RCURLY, "} expected");
+        
+        //main in service needs to be defined
+        if ( internal_main == null )
+            throwException( "You must specify a main for internal service " + serviceName );
+      
+        Program internalServiceProgram = new Program(getContext());
+        
+        // copy children of parent to embedded service
+        for ( OLSyntaxNode child : program.children() ) {
+            if ( child instanceof TypeInlineDefinition ||
+                child instanceof InterfaceDefinition ||
+                child instanceof OutputPortInfo ||
+                child instanceof EmbeddedServiceNode )
```

Comment:
This does not look like the right way of implementing the first requirement in Semantics of #27 .
It's going to duplicate embedded services, which could be a problem in case of sub-services written in Jolie (and even Java, although many are harmless, like Console).

What we should do is implement some mechanism for the internal service to wait for the bindings for the output ports at runtime. This doesn't look too easy, we need to think of the lifecycle defined in:

https://github.com/jolie/jolie/blob/master/jolie/src/jolie/process/InitDefinitionProcess.java

Embedded services are loaded before the init procedure starts, so the internal service can't wait for the parent's init to end before "copying over" the bindings for the output ports or we'll create a deadlock.